src/main/java/pl/magisterka/dominikszojda/mappers/DogMapper.java

```diff
@@ -0,0 +1,13 @@
+package pl.magisterka.dominikszojda.mappers;
+
+import org.mapstruct.Mapper;
+import pl.magisterka.dominikszojda.entity.DogEntity;
+import pl.magisterka.dominikszojda.request.DogCreateRequest;
+import pl.magisterka.dominikszojda.respons.DogResponse;
+
+@Mapper(componentModel = "spring")
+public interface DogMapper {
```

Comment:
Dobrą praktyką jest dodać do `@Mapper` parametr `unmappedTargetPolicy = ReportingPolicy.ERROR` by MapStruct zgłaszał błąd kompilacji jeżeli zapomnisz coś zmappować. Pozwala to wyeliminować część potencjalnie trudnych do wykrycia błedów, które mogą powstać gdy zaczniesz dodawać lub zmieniać strukturę encji/dto. 

```suggestion
@Mapper(
    componentModel = "spring",
    unmappedTargetPolicy = ReportingPolicy.ERROR)
public interface DogMapper {
```

Jeżeli zacznie ci przybywać coraz więcej takich parametrów konfiguracyjnych, dobrym pomysłem jest zrobić "config":

```java
// DogMapper.java
@Mapper(config = MyMapperConfig.class)
public interface DogMapper { ... }

// MyMapperConfig.java
@MapperConfig(
    componentModel = "spring",
    unmappedTargetPolicy = ReportingPolicy.ERROR,
    nullValueIterableMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT
    /* itd... */)
)
public interface MyMapperConfig {}
```